lib/slack-handler.js

```diff
@@ -6,16 +6,24 @@ var HELPERS = {
   getIrcChannelName: function getIrcChannelName(context, slackChannel) {
     var channelConfig = {};
     try {
-      if (slackChannel.purpose && slackChannel.purpose.value) {
+      if (slackChannel.purpose && slackChannel.purpose.value &&
+          HELPERS.detectJsonString(slackChannel.purpose.value)) {
```

Comment:
The JSON detection could probably be part of the `parseConfig` method, so that it would apply to all configuration reading.